### ISSUE TITLE
Get accessibility image tagging working

### ISSUE THREAD

**Issue by seananderson** [pbs-assess/csasdowntex#44]
I think @andrew-edwards (and others?) might have already worked on this. There are a number of options including tooltip packages.

If it is possible then we should add something to the wiki about it.

**Comment by andrew-edwards** [comment]
The info for using tooltips is still on the readme at

https://github.com/pbs-assess/csas-style

and I'd also mentioned the accsupp package in https://github.com/pbs-assess/csas-style/issues/3  but hadn't tried it. @grinnellm referenced that issue wrt herring in:

https://github.com/grinnellm/HerringAssess/issues/40

The POP document I mentioned in https://github.com/pbs-assess/csas-style/issues/3 was probably just .tex (or maybe .Snw), not full knitr or .Rmd. 

Basically you want alternate text for each figure, that just says "Figure X" when you hover a mouse over it. That (and the caption) should get read out loud automatically when the screen reader reads out loud. Without the alternate text I think some screen readers get stuck.

**Comment by seananderson** [comment]
The tooltip option seems to read correctly through a screen reader although it gets flagged as incorrect alternate text in Acrobat Pro. As in the past, we figure that we can argue that this is OK. @cgrandin implemented a regular expression to wrap all includegraphics commands in this.

We tried every other possibility and can't find anything else that works.

**Comment by ricardd** [comment]
@seananderson and @cgrandin , how could the the pdftooltip chunk added by gsub in utils.R be modified to also include the figure number? It was asked by my regional CSAS coordinator for the alt text to be the actual figure number. It is a pretty minor point that I can probably argue against (just "Figure" is good enough), but if you can concoct an elegant solution it would also be fine.

I am wrapping my head around this:

https://stackoverflow.com/questions/4213800/is-there-something-like-a-counter-variable-in-regular-expression-replace

**Comment by seananderson** [comment]
I assume the ref: label bit will render into the number. If so, you’ll need to write a regular expression to grab it from a nearby label. Seems a bit hard to do reliably since they could be in a number of places. Another option would be to grab them all for the whole document and then insert them in order. Maybe that would work better?

**Comment by cgrandin** [comment]
The fix I did will work for figures as long as they are monotonically increasing.. i.e. if there are appendix figures like A.1 A.2 etc this will break. It's a starting point and we can just revert the commit - it doesn't matter to me either way

**Comment by seananderson** [comment]
My latest commit should add the `ref:` label for each tooltip. I think this will work for all situations because of the standardized formatting of Pandoc, but if you notice any problems let us know. If it fails to add the figure label for tooltip numbers, it will revert to the old method of just having "Figure" and issue a warning to the user.

**Comment by ricardd** [comment]
@seananderson thanks, that works fine for me.

**Comment by cgrandin** [comment]
Thanks @seananderson, I missed all the fun after starting as I had to go to the dentist! This seems like a good fix. Now to see if I can break it with tests ... haha

**Comment by andrew-edwards** [comment]
Nice work guys - I'm just catching up and was just wondering if this ever got automatically incorporated. Good job...

**Comment by seananderson** [comment]
Yes, it's incorporated: https://github.com/pbs-assess/csasdown/blob/f2f0cfb99e93415c42534b717364727df25d44cf/R/utils.R#L228-L253
